src/components/ArticleBodyById.jsx

```diff
@@ -13,42 +18,84 @@ function ArticleBodyById() {
 
   const [votes, setVotes] = useState();
 
+  const [commentInput, setCommentInput] = useState("");
+
+  const [newCommentExists, setNewCommentExists] = useState(1);
+
   useEffect(() => {
     setIsLoading(true);
     fetchArticleById(id).then((data) => {
       setArticleById(data);
       setVotes(data.votes);
       setIsLoading(false);
     });
+  }, []);
 
+  useEffect(() => {
     fetchCommentsById(id).then((data) => {
       setCommentsById(data.comments);
-      setIsLoading(false);
     });
-  }, []);
+  }, [newCommentExists]);
 
   function handleLike(event) {
+    event.preventDefault();
     const incVote = Number(event.target.value);
     setVotes((currentVotes) => currentVotes + incVote);
     updateVotes(id, incVote);
   }
 
+  function handleSubmit(event) {
+    event.preventDefault();
+    if (commentInput) {
+      postComment(id, commentInput, user).then((response) => {
+        if (response.status === 201) {
+          setNewCommentExists((curr) => {
+            return curr + 1;
```

Comment:
You don't need the if statement here. If you made it into the .then block then the post request is successful 😄